pkg/store/bucket.go

```diff
@@ -1284,6 +1302,15 @@ func newBucketBlock(
 		enablePostingsCompression: enablePostingsCompression,
 	}
 
+	// Translate the block's labels and inject the block ID as a label
+	// to allow to match blocks also by ID.
+	b.matcherLabels = labels.FromMap(meta.Thanos.Labels)
```

Comment:
```suggestion
	b.extLabelsAndBlockID = labels.FromMap(meta.Thanos.Labels)
```

no better idea :shrug: 
Maybe `relabelLabels` would do to say that we are it's the same as doing relabel